rllib/utils/tests/test_check_env.py

```diff
@@ -77,39 +84,108 @@ def test_step(self):
         env.step = step
         error = ".*The observation collected from env.step.*"
         with pytest.raises(ValueError, match=error):
-            check_gym_environments(env)
+            check_env(env)
 
         # check reset that returns obs of incorrect type fails
         step = MagicMock(return_value=(float(1), 5, True, {}))
         env.step = step
         with pytest.raises(ValueError, match=error):
-            check_gym_environments(env)
+            check_env(env)
 
         # check step that returns reward of non float/int fails
         step = MagicMock(return_value=(1, "Not a valid reward", True, {}))
         env.step = step
         error = ("Your step function must return a reward that is integer or "
                  "float.")
         with pytest.raises(AssertionError, match=error):
-            check_gym_environments(env)
+            check_env(env)
 
         # check step that returns a non bool fails
         step = MagicMock(
             return_value=(1, float(5), "not a valid done signal", {}))
         env.step = step
         error = "Your step function must return a done that is a boolean."
         with pytest.raises(AssertionError, match=error):
-            check_gym_environments(env)
+            check_env(env)
 
         # check step that returns a non dict fails
         step = MagicMock(
             return_value=(1, float(5), True, "not a valid env info"))
         env.step = step
         error = "Your step function must return a info that is a dict."
         with pytest.raises(AssertionError, match=error):
-            check_gym_environments(env)
+            check_env(env)
+        del env
+
+
+class TestCheckMultiAgentEnv():
+    @pytest.fixture(autouse=True)
+    def inject_fixtures(self, caplog):
+        caplog.set_level(logging.CRITICAL)
+
+    def test_check_env_not_correct_type_error(self):
+        env = RandomEnv()
+        with pytest.raises(ValueError, match="The passed env is not"):
+            check_multiagent_environments(env)
         del env
 
+    def test_check_env_reset_incorrect_error(self):
+        reset = MagicMock(return_value=5)
+        env = make_multi_agent("CartPole-v1")({"num_agents": 2})
+        env.reset = reset
+        with pytest.raises(
+                ValueError, match="The observation collected from "):
+            check_env(env)
+
+    def test_check_incorrect_space_contains_functions_error(self):
+        def bad_contains_function(self, x):
```

Comment:
you know, this is not the best kind of mock.
what we are mocking here is to directly return error, and unit test if we see an error.
it may be better to mock an action instead that doesn't fall in the action space, and double check that we fires the correct error from the actual code.

Reply:
I can mock an out of bounds action

Reply:
added both